atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/qos/ThriftQueryWeighers.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.cassandra.thrift.CqlResult;
+import org.apache.cassandra.thrift.KeyRange;
+import org.apache.cassandra.thrift.KeySlice;
+import org.apache.cassandra.thrift.Mutation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Suppliers;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraClient;
+import com.palantir.atlasdb.qos.ImmutableQueryWeight;
+import com.palantir.atlasdb.qos.QosClient;
+import com.palantir.atlasdb.qos.QueryWeight;
+
+public final class ThriftQueryWeighers {
+
+    private static final Logger log = LoggerFactory.getLogger(CassandraClient.class);
+
+    @VisibleForTesting
+    private static final int ESTIMATED_NUM_BYTES_PER_ROW = 100;
+    static final QueryWeight DEFAULT_ESTIMATED_WEIGHT = ImmutableQueryWeight.builder()
+            .numBytes(ESTIMATED_NUM_BYTES_PER_ROW)
+            .numDistinctRows(1)
+            .timeTakenNanos(TimeUnit.MILLISECONDS.toNanos(2))
+            .build();
+
+    static final QueryWeight ZERO_ESTIMATED_WEIGHT = ImmutableQueryWeight.builder()
+            .numBytes(0)
+            .numDistinctRows(1)
+            .timeTakenNanos(TimeUnit.MILLISECONDS.toNanos(1))
+            .build();
+
+    private ThriftQueryWeighers() { }
+
+    static QosClient.QueryWeigher<Map<ByteBuffer, List<ColumnOrSuperColumn>>> multigetSlice(List<ByteBuffer> keys,
+            boolean zeroEstimate) {
+        return zeroEstimate
+                ? readWeigherWithZeroEstimate(ThriftObjectSizeUtils::getApproximateSizeOfColsByKey, Map::size,
+                keys.size())
+                : readWeigher(ThriftObjectSizeUtils::getApproximateSizeOfColsByKey, Map::size, keys.size());
+    }
+
+    static QosClient.QueryWeigher<List<KeySlice>> getRangeSlices(KeyRange keyRange, boolean zeroEstimate) {
+        return zeroEstimate
+                ? readWeigherWithZeroEstimate(ThriftObjectSizeUtils::getApproximateSizeOfKeySlices, List::size,
+                keyRange.count)
+                : readWeigher(ThriftObjectSizeUtils::getApproximateSizeOfKeySlices, List::size, keyRange.count);
+    }
+
+    static QosClient.QueryWeigher<ColumnOrSuperColumn> get(boolean zeroEstimate) {
+        return zeroEstimate
+                ? readWeigherWithZeroEstimate(ThriftObjectSizeUtils::getColumnOrSuperColumnSize, ignored -> 1, 1)
+                : readWeigher(ThriftObjectSizeUtils::getColumnOrSuperColumnSize, ignored -> 1, 1);
+    }
+
+    static final QosClient.QueryWeigher<CqlResult> EXECUTE_CQL3_QUERY =
+            // TODO(nziebart): we need to inspect the schema to see how many rows there are - a CQL row is NOT a
+            // partition. rows here will depend on the type of query executed in CqlExecutor: either (column, ts) pairs,
+            // or (key, column, ts) triplets
+            // Currently, transaction or metadata table queries dont use the CQL executor,
```

Comment:
I added a zero estimation for other transaction or metadata queries. Since its not using the cql executor we can get away with this for now.